sql/src/main/java/org/apache/druid/sql/calcite/planner/JoinHint.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.sql.calcite.planner;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import org.apache.druid.java.util.common.IAE;
+import org.apache.druid.query.JoinAlgorithm;
+
+import java.util.Arrays;
+
+public enum JoinHint
+{
+  SORT_MERGE("sort_merge") {
+    @Override
+    public JoinAlgorithm getJoinAlgorithm()
+    {
+      return JoinAlgorithm.SORT_MERGE;
+    }
+  },
+  BROADCAST("broadcast") {
+    @Override
+    public JoinAlgorithm getJoinAlgorithm()
+    {
+      return JoinAlgorithm.BROADCAST;
+    }
+  };
+
+  private final String id;
+
+  JoinHint(String id)
+  {
+    this.id = id;
+  }
+
+  @JsonCreator
+  public static JoinHint fromString(final String id)
+  {
```

Comment:
I don't think this should be serializable by jackson; if it gets on the wire it should be a `JoinAlgorithm` already

Reply:
I don't think this `JoinHint` should be serialized
if its not in use I think it would be better to remove it